std/numeric.d

```diff
@@ -2987,15 +2987,18 @@ if (isIntegral!T)
 // This overload is for non-builtin numerical types like BigInt or
 // user-defined types.
 /// ditto
-T gcd(T)(T a, T b)
+auto gcd(T)(T _a, T _b)
 if (!isIntegral!T &&
         is(typeof(T.init % T.init)) &&
         is(typeof(T.init == 0 || T.init > 0)))
 {
     import std.algorithm.mutation : swap;
 
+    Unqual!T a = _a;
+    Unqual!T b = _b;
+
```

Comment:
Can you `static if` this so you don't make copies if you don't have to?  If T is not const, just `alias a = _a`?

Reply:
Other option: use `const T` for the params and then always make a mutable copy. This would prevent all the 'redundant' const/immutable instantiations, another (common) problem of such public templates. Or just forward to the mutable version as integer gcd above: https://github.com/dlang/phobos/blob/a394c1348b4b73765bd0bbf4d08b0a8548c5778b/std/numeric.d#L2936-L2939

Reply:
The main problem with the `const T` idea over the `alias` is it pessimizes all code: the copy will always happen. There's also a theoretical issue of a type `T` such that `const(T)` can't be converted to `T`.

Reply:
So you've changed your mind?